core/src/main/scala/pureconfig/DerivedReaders.scala

```diff
@@ -198,6 +232,32 @@ trait DerivedReaders1 {
     }
   }
 
+  implicit final lazy val hNilConfigReader: ConfigReader[HNil] =
+    new ConfigReader[HNil] {
+      def from(cv: ConfigValue): Either[ConfigReaderFailures, HNil] = {
+        cv match {
+          case cl: ConfigList if cl.size == 0 => Right(HNil)
+          case cl: ConfigList => fail(WrongSizeList(0, cl.size, ConfigValueLocation(cv), ""))
+          case other => fail(WrongType(other.valueType, Set(ConfigValueType.LIST), ConfigValueLocation(other), ""))
+        }
+      }
+    }
+
+  implicit final def hConsConfigReader[H, T <: HList](implicit hr: Derivation[Lazy[ConfigReader[H]]], tr: Derivation[Lazy[ConfigReader[T]]], tl: HKernelAux[T]): ConfigReader[H :: T] =
```

Comment:
Note that the recursive part of the derivation should not be wrapped in `Derivation`. If you do so, when you try to derive for example a reader for `String :: Custom :: HNil` you'll get a message like this:

```
Could not derive a ConfigReader instance for String :: Int :: Custom :: HNil, because:
  - could not derive a ConfigReader instance for Int :: Custom :: HNil, because:
    - could not derive a ConfigReader instance for Custom :: HNil, because:
      - could not derive a ConfigReader instance for Custom
```

When you probably want just:

```
Could not derive a ConfigReader instance for String :: Int :: Custom :: HNil, because:
  - could not derive a ConfigReader instance for Custom
```